client/asset/dcr/dcr.go

```diff
@@ -919,6 +919,40 @@ func (dcr *ExchangeWallet) FundingCoins(ids []dex.Bytes) (asset.Coins, error) {
 	if len(notFound) == 0 {
 		return coins, nil
 	}
+
+	// Check locked outputs for not found coins.
+	lockedOutputs, err := dcr.lockedOutputs()
+	if err != nil {
+		return nil, err
+	}
+	for _, output := range lockedOutputs {
+		txHash, err := chainhash.NewHashFromStr(output.Txid)
+		if err != nil {
+			return nil, fmt.Errorf("error decoding txid from rpc server %s: %v", output.Txid, err)
+		}
+		pt := newOutPoint(txHash, output.Vout)
+		if notFound[pt] {
+			txOut, err := dcr.node.GetTxOut(txHash, output.Vout, true)
+			if err != nil {
+				return nil, fmt.Errorf("gettxout error for locked output %v: %v", pt.String(), err)
+			}
+			var address string
+			if len(txOut.ScriptPubKey.Addresses) > 0 {
+				address = txOut.ScriptPubKey.Addresses[0]
+			}
+			coin := newOutput(dcr.node, txHash, output.Vout, toAtoms(output.Amount), output.Tree, nil)
```

Comment:
The return type for `listunspent` has a `RedeemScript` field that was used here. Since locked outputs were once returnable by `listunspent`, I sort to find out how dcrwallet populates the `RedeemScript` value for `listunspent` results and found out that dcrwallet does not:
https://github.com/decred/dcrwallet/blob/7806b30f9b9f9ad516342af6706b9d2df2e896e8/wallet/wallet.go#L3283-L3293

Is that a cause for concern? Looking through dcr's usage of funding coins in the `Swap` method, I don't see a use for the redeem script, so we're probably safe.

Reply:
@jrick we chatted in matrix, but it does look like we'd like redeem script set.